src/field/pokemon.ts

```diff
@@ -933,7 +936,9 @@ export default abstract class Pokemon extends Phaser.GameObjects.Container {
     if (this.metBiome === -1 && !this.scene.gameMode.isFreshStartChallenge() && !this.scene.gameMode.isDaily) {
       levelMoves = this.getUnlockedEggMoves().concat(levelMoves);
     }
-    return levelMoves.filter(lm => !this.moveset.some(m => m?.moveId === lm));
+    levelMoves = this.usedTMs?.length? this.usedTMs.filter(m => !levelMoves.includes(m)).concat(levelMoves) : levelMoves;
```

Comment:
```suggestion
    if (Array.isArray(this.usedTMs) && this.usedTMs.length !== 0) {
      levelMoves = this.usedTMs.filter(m => !levelMoves.includes(m)).concat(levelMoves);
    }
```
Much safer and readable to make this an `if` rather than an extra ternary that, in the `false` case, just does an extra `levelMoves = levelMoves` assignment.